### DISCUSSION TITLE
marp-cli error when using markdown-it-graphiz

### DISCUSSION THREAD

**Discussion by alanlivio** [marp-team/marp#434] [Q&A] [answered]
Hello. I followed this [guide to use plantuml](https://github.com/orgs/marp-team/discussions/219) from @yhatt to use a graphiz. But I get the error bellow.
Any suggestions? 
Thank you in advance.
Best, 
Alan

engine.js file
```js
const { Marp } = require('@marp-team/marp-core')
module.exports = {
    engine: opts => new Marp(opts).use(require('markdown-it-graphviz'))
}
```

package.json file
```
{
  "dependencies": {
    "@marp-team/marp-cli": "^2.5.0",
    "@marp-team/marp-core": "0.13.1",
    "@marp-team/marpit": "1.4.0",
    "markdown-it-graphviz": "^1.1.2"
  }
}
``` 

graphviz.md file
````markdown
---
marp: true
---

```graphviz
strict graph {
  a -- b
  a -- y
  b -- a [color=blue]
}
```
````

command line
```bash
npx marp --engine ./engine.js graphviz.md
```
![image](https://user-images.githubusercontent.com/604808/233430386-1108035d-c876-4545-9457-9a870440e5eb.png)

**Comment by yhatt** [comment]
Multipost: See #433

**Reply by alanlivio** [reply]
Thanks @yhatt. I tried the guide from #433, but it didn't work. Please take a look at the code below.

package.json
```json
{
  "dependencies": {
    "@marp-team/marp-cli": "^2.5.0",
    "@marp-team/marp-core": "^3.6.0",
    "@marp-team/marpit": "^2.4.2",
    "markdown-it-graphviz": "latest"
  }
}
```
marp.config.js
```js
const { Marp } = require('@marp-team/marp-core')

module.exports = {
    engine: opts => new Marp(opts).use(require('markdown-it-graphviz'))
}
```

I used the same graphviz.md then ran `marp graphviz.md`.

![image](https://user-images.githubusercontent.com/604808/233589515-f28655ca-02a4-46ac-bd66-e1e9284291d8.png)

**Reply by yhatt** [reply, marked as answer]
A (probably not maintained) markdown-it plugin `markdown-it-graphviz` you are trying to use is broken: https://github.com/aiyoudiao/markdown-it-graphviz/issues/2 (Chinese)

It tries to load a script written in ES Module [`./common/viz.es-changed.js`](https://github.com/aiyoudiao/markdown-it-graphviz/blob/master/common/viz.es-changed.js) from CommonJS script [`index.js`](https://github.com/aiyoudiao/markdown-it-graphviz/blob/master/index.js), but [Node.js cannot import ESM script from CJS through `require()`](https://nodejs.org/api/esm.html#require).  

In regards to this, there is no way to do in user-side. Try another markdown-it plugin that supports graphviz correctly.

**Reply by alanlivio** [reply]
Thanks @yhatt. It worked with [this other package](https://www.npmjs.com/package/markdown-it-graphviz-exec).
I put my used code below.

package.json
```json
{
  "dependencies": {
    "@marp-team/marp-cli": "^2.5.0",
    "@marp-team/marp-core": "^3.6.0",
    "@marp-team/marpit": "^2.4.2",
    "markdown-it-graphviz-exec": "latest"
  }
}
```
marp.config.js
```js
const { Marp } = require('@marp-team/marp-core')

module.exports = {
    engine: opts => new Marp(opts).use(require('markdown-it-graphviz-exec'))
}
```

But I used a slightly different markdown following the [package](https://www.npmjs.com/package/markdown-it-graphviz-exec).

```markdown
---
marp: true
---

@graphviz_open
strict graph {
  a -- b
  a -- y
  b -- a [color=blue]
}
@graphviz_close
```

![image](https://user-images.githubusercontent.com/604808/233674595-9121c89f-96bb-4a3d-90d1-3eb7a3e43e44.png)
